doc: Add ORDER BY documentation (#1524)
Enalean/tuleap-documentation-en: languages/en/user-guide/tql.rst

To provide both condition, you can use ``AND`` between them. There is no restriction for the order of the conditions.

> TQL ``ORDER BY`` syntax allow you to sort artifacts according to a field. The allowed fields are the same as for the condition with the same restrictions.

Reviewer: issue: It's not exactly the same fields: sorting is only allowed on list fields with a single possible value (selectbox and radio buttons, NOT open lists, multi-selectbox or checkbox)

Author: I completely forgot this point

> 

Reviewer: suggestion: we should also mention that ordering does not allow mixing date and datetime fields

Author: Done

You must provide the direction of the ordering:
 * ``ASC`` or ``ASCENDING`` from smallest to largest
 * ``DESC`` or ``DESCENDING`` from largest to smallest

> If you do not provide an ``ORDER BY`` to your query, it will default to ``ORDER BY @id DESCENDING``.

Reviewer: praise: :+1:

To provide both condition, you can use ``AND`` between them. There is no restriction for the order of the conditions.

> TQL ``ORDER BY`` syntax allow you to sort artifacts according to a field.

Author: Done

> There is some restrictions on the fields you can use. As for the condition their types must be compatible, date and date with time fields are not compatible. 

Author: Done

> For list field only selectbox and radio buttons are allowed. All the ``@`` fields are allowed.

Author: Done

To provide both condition, you can use ``AND`` between them. There is no restriction for the order of the conditions.

TQL ``ORDER BY`` syntax allows you to sort artifacts on a single field. Ordering on more than one criteria is not possible.

There are some restrictions on the fields you can use. As for the condition (``WHERE``) their types must be compatible. Date and date with time fields are not compatible together.
For list fields, only selectbox and radio buttons are allowed. All the ``@`` fields and semantics valid in conditions (``WHERE``) are allowed.

You must provide the direction of the ordering:
 * ``ASC`` or ``ASCENDING`` from smallest to largest
 * ``DESC`` or ``DESCENDING`` from largest to smallest

> User list and user group list fields are compared on displayed value. For user lists it means that result depends on your preference on user display (login vs realname). 

Reviewer: ```suggestion
User list and user group list fields are compared on displayed value. For user lists, it means that the ordering depends on your preference on user display (login, real name or both).
```

Author: Done

> For user groups like project members or project admins, the sort is done on their translated name and so result depends on your language.

Reviewer: ```suggestion
For user groups like Project members or Project administrators, the sort is done on their translated name and so the ordering depends on your language.
```

Author: Done
